File: 07-ci-cd/scripts/step-summary.ts
#!/usr/bin/env ts-node
// Read report.json and print a GitHub Actions step summary (Markdown).
import * as fs from "fs";

const path = process.argv[2] ?? "report.json";
const report = JSON.parse(fs.readFileSync(path, "utf-8"));
const s = report.summary;

const icon = s.failed === 0 ? "✅" : "❌";
const severityOrder: Record<string, number> = { critical: 0, high: 1, medium: 2, low: 3 };

const lines: string[] = [
  `## ${icon} Test Results`,
  "",
  "| Total | Passed | Failed | Bugs |",
  "|-------|--------|--------|------|",
  `| ${s.total} | ${s.passed} | ${s.failed} | ${s.bugs} |`,
  "",
  "### Scenarios",
  "",
  "| ID | Title | Result |",
  "|----|-------|--------|",
];

for (const r of [...report.results].sort((a: any, b: any) =>
  a.scenarioId.localeCompare(b.scenarioId)
)) {
  lines.push(`| ${r.scenarioId} | ${r.title} | ${r.passed ? "✅ PASS" : "❌ FAIL"} |`);
}

if (report.findings?.length) {
  lines.push(
    "",
    "### Findings",
    "",
    "| Severity | Classification | Scenario | Description |",
    "|----------|---------------|----------|-------------|"
  );
  const sorted = [...report.findings].sort(
    (a: any, b: any) => severityOrder[a.severity] - severityOrder[b.severity]
  );
  for (const f of sorted) {
    const desc = f.description.length > 80 ? f.description.slice(0, 80) + "…" : f.description;
    lines.push(`| ${f.severity.toUpperCase()} | ${f.classification} | ${f.scenarioId} | ${desc} |`);
  }
}

console.log(lines.join("\n"));

File: 07-ci-cd/scripts/step-summary.py
#!/usr/bin/env python3
"""Read report.json and print a GitHub Actions step summary (Markdown)."""
import json
import sys

path = sys.argv[1] if len(sys.argv) > 1 else "report.json"

with open(path) as f:
    report = json.load(f)

summary = report["summary"]
icon = "✅" if summary["failed"] == 0 else "❌"

lines = [
    f"## {icon} Test Results",
    "",
    f"| Total | Passed | Failed | Bugs |",
    f"|-------|--------|--------|------|",
    f"| {summary['total']} | {summary['passed']} | {summary['failed']} | {summary['bugs']} |",
    "",
    "### Scenarios",
    "",
    "| ID | Title | Result |",
    "|----|-------|--------|",
]

for r in sorted(report["results"], key=lambda x: x["scenarioId"]):
    status = "✅ PASS" if r["passed"] else "❌ FAIL"
    lines.append(f"| {r['scenarioId']} | {r['title']} | {status} |")

if report["findings"]:
    severity_order = {"critical": 0, "high": 1, "medium": 2, "low": 3}
    lines += [
        "",
        "### Findings",
        "",
        "| Severity | Classification | Scenario | Description |",
        "|----------|---------------|----------|-------------|",
    ]
    findings = sorted(report["findings"], key=lambda f: severity_order[f["severity"]])
    for f in findings:
        desc = f["description"][:80] + ("…" if len(f["description"]) > 80 else "")
        lines.append(
            f"| {f['severity'].upper()} | {f['classification']} | {f['scenarioId']} | {desc} |"
        )

print("\n".join(lines))
